pkg/logs/input/tailer/scanner.go

```diff
@@ -96,9 +98,18 @@ func (s *Scanner) run() {
 // The Scanner needs to stop that previous tailer,
 // and start a new one for the new file.
 func (s *Scanner) scan() {
-	for _, source := range s.sources {
-		tailer := s.tailers[source.Path]
-		f, err := os.Open(source.Path)
+	files := s.fileProvider.FilesToTail()
```

Comment:
If the implementation of `FilesToTail` is not guaranteed to be constant when using wildcards, won't multiple calls to `scan` result in the agent going over the limit of number of files to tail ?

Reply:
Good catch, you're right 👍

Reply:
You're right but just for a small amount of time because we only keep running the tailers matching `FilesToTail`, all the others are stopped and garbage-collected.
We could still add another protection though to strickly prevent the number of tailers from going above the limit, WDYT ?

Reply:
It might be worth it to mention this if we don't the change though

Reply:
This is enough to me personally.. however if we guarantee our customers we won't open more than `x` files I'm not sure it's a smart move to go over it, even for a short time. I'm not sure what's the agent policy on things like this, people from croissant might have better suggestions to provide.

Reply:
done